[AUTOCUT] Integration Test Failed for k-NN-2.19.0

### Integration Test Failed for version 2.19.0. See the specifications below:

#### Details

| Platform | Dist | Arch | Dist Build No. | RC | Test Report | Workflow Run | Failing tests |
|----------|------|------|----------------|----|-------------|--------------|---------------|
| linux | tar | x64 | 10777 | 3 | https://ci.opensearch.org/ci/dbc/integ-test/2.19.0/10777/linux/x64/tar/test-results/9320/integ-test/test-report.yml | https://build.ci.opensearch.org/job/integ-test/9320/display/redirect | [Check metrics](https://metrics.opensearch.org/_dashboards/app/dashboards?security_tenant=global#/view/21aad140-49f6-11ef-bbdd-39a9b324a5aa?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-30d,to:now))&_a=(description:'OpenSearch%20Release%20Build%20and%20Integration%20Test%20Results',expandedPanelId:ddafb9c5-2d35-482a-9c61-1ba78b67f406,filters:!(('':(store:appState),meta:(alias:!n,disabled:!f,index:d90d2ba0-8fe0-11ef-a168-f19b1bbc360c,key:version,negate:!f,params:(query:'2.19.0'),type:phrase),query:(match_phrase:(version:'2.19.0'))),('':(store:appState),meta:(alias:!n,disabled:!f,index:d90d2ba0-8fe0-11ef-a168-f19b1bbc360c,key:component,negate:!f,params:(query:k-NN),type:phrase),query:(match_phrase:(component:k-NN)))),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),query:(language:kuery,query:''),timeRestore:!t,title:'OpenSearch%20Release%20Build%20and%20Integration%20Test%20Results',viewMode:view)) |
| linux | tar | x64 | 10777 | 3 | https://ci.opensearch.org/ci/dbc/integ-test/2.19.0/10777/linux/x64/tar/test-results/9319/integ-test/test-report.yml | https://build.ci.opensearch.org/job/integ-test/9319/display/redirect | [Check metrics](https://metrics.opensearch.org/_dashboards/app/dashboards?security_tenant=global#/view/21aad140-49f6-11ef-bbdd-39a9b324a5aa?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-30d,to:now))&_a=(description:'OpenSearch%20Release%20Build%20and%20Integration%20Test%20Results',expandedPanelId:ddafb9c5-2d35-482a-9c61-1ba78b67f406,filters:!(('':(store:appState),meta:(alias:!n,disabled:!f,index:d90d2ba0-8fe0-11ef-a168-f19b1bbc360c,key:version,negate:!f,params:(query:'2.19.0'),type:phrase),query:(match_phrase:(version:'2.19.0'))),('':(store:appState),meta:(alias:!n,disabled:!f,index:d90d2ba0-8fe0-11ef-a168-f19b1bbc360c,key:component,negate:!f,params:(query:k-NN),type:phrase),query:(match_phrase:(component:k-NN)))),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),query:(language:kuery,query:''),timeRestore:!t,title:'OpenSearch%20Release%20Build%20and%20Integration%20Test%20Results',viewMode:view)) |
| linux | tar | x64 | 10777 | 3 | https://ci.opensearch.org/ci/dbc/integ-test/2.19.0/10777/linux/x64/tar/test-results/9321/integ-test/test-report.yml | https://build.ci.opensearch.org/job/integ-test/9321/display/redirect | [Check metrics](https://metrics.opensearch.org/_dashboards/app/dashboards?security_tenant=global#/view/21aad140-49f6-11ef-bbdd-39a9b324a5aa?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-30d,to:now))&_a=(description:'OpenSearch%20Release%20Build%20and%20Integration%20Test%20Results',expandedPanelId:ddafb9c5-2d35-482a-9c61-1ba78b67f406,filters:!(('':(store:appState),meta:(alias:!n,disabled:!f,index:d90d2ba0-8fe0-11ef-a168-f19b1bbc360c,key:version,negate:!f,params:(query:'2.19.0'),type:phrase),query:(match_phrase:(version:'2.19.0'))),('':(store:appState),meta:(alias:!n,disabled:!f,index:d90d2ba0-8fe0-11ef-a168-f19b1bbc360c,key:component,negate:!f,params:(query:k-NN),type:phrase),query:(match_phrase:(component:k-NN)))),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),query:(language:kuery,query:''),timeRestore:!t,title:'OpenSearch%20Release%20Build%20and%20Integration%20Test%20Results',viewMode:view)) |

Check out test report manifest linked above for steps to reproduce, cluster and integration test failure logs. For additional information checkout the [wiki](https://github.com/opensearch-project/opensearch-build/wiki/Testing-the-Distribution) and [OpenSearch Metrics Dashboard](https://metrics.opensearch.org/_dashboards/app/dashboards#/view/21aad140-49f6-11ef-bbdd-39a9b324a5aa).

Tagging the release owners to take a look @kotwanikunal

```
EPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.RestoreSnapshotIT.testKnnSettingIsModifiable_whenRestore_thenSuccess" -Dtests.seed=6F3E1A3C9D9E53F4 -Dtests.security.manager=false -Dtests.locale=bg-BG -Dtests.timezone=Japan -Druntime.java=21



REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.RestoreSnapshotIT.testKnnSettingCanBeIgnored_whenRestore_thenSuccess" -Dtests.seed=6F3E1A3C9D9E53F4 -Dtests.security.manager=false -Dtests.locale=bg-BG -Dtests.timezone=Japan -Druntime.java=21



REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.RestoreSnapshotIT.testKnnSettingCannotBeIgnored_whenRestore_thenFailure" -Dtests.seed=6F3E1A3C9D9E53F4 -Dtests.security.manager=false -Dtests.locale=bg-BG -Dtests.timezone=Japan -Druntime.java=21



REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.RestoreSnapshotIT.testKnnSettingIsUnmodifiable_whenRestore_thenFailure" -Dtests.seed=6F3E1A3C9D9E53F4 -Dtests.security.manager=false -Dtests.locale=bg-BG -Dtests.timezone=Japan -Druntime.java=21





Suite: Test class org.opensearch.knn.index.RestoreSnapshotIT

  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.RestoreSnapshotIT.testKnnSettingIsModifiable_whenRestore_thenSuccess" -Dtests.seed=6F3E1A3C9D9E53F4 -Dtests.security.manager=false -Dtests.locale=bg-BG -Dtests.timezone=Japan -Druntime.java=21

  2> java.lang.AssertionError: 

    Expected: a collection with size <1>

         but: collection size was <0>

        at __randomizedtesting.SeedInfo.seed([6F3E1A3C9D9E53F4:3B133CC9D7FCAAF9]:0)

        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)

        at org.junit.Assert.assertThat(Assert.java:964)

        at org.junit.Assert.assertThat(Assert.java:930)

        at org.opensearch.knn.KNNRestTestCase.setupSnapshotRestore(KNNRestTestCase.java:1997)

        at org.opensearch.knn.index.RestoreSnapshotIT.setUp(RestoreSnapshotIT.java:29)

  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.RestoreSnapshotIT.testKnnSettingCanBeIgnored_whenRestore_thenSuccess" -Dtests.seed=6F3E1A3C9D9E53F4 -Dtests.security.manager=false -Dtests.locale=bg-BG -Dtests.timezone=Japan -Druntime.java=21

  2> java.lang.AssertionError: 

    Expected: a collection with size <1>

         but: collection size was <0>

        at __randomizedtesting.SeedInfo.seed([6F3E1A3C9D9E53F4:D89A96940FFDC5C9]:0)

        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)

        at org.junit.Assert.assertThat(Assert.java:964)

        at org.junit.Assert.assertThat(Assert.java:930)

        at org.opensearch.knn.KNNRestTestCase.setupSnapshotRestore(KNNRestTestCase.java:1997)

        at org.opensearch.knn.index.RestoreSnapshotIT.setUp(RestoreSnapshotIT.java:29)
```

--- a/build.gradle
+++ b/build.gradle
@@ -373,6 +373,7 @@ integTest {
     systemProperty 'tests.security.manager', 'false'
     systemProperty 'java.io.tmpdir', opensearch_tmp_dir.absolutePath
     systemProperty "java.library.path", "$rootDir/jni/release"
+    systemProperty "tests.path.repo", "${buildDir}/testSnapshotFolder"
     // allows integration test classes to access test resource from project root path
     systemProperty('project.root', project.rootDir.absolutePath)
 
@@ -438,6 +439,9 @@ testClusters.integTest {
         }
     }
     systemProperty("java.library.path", "$rootDir/jni/release")
+    final testSnapshotFolder = file("${buildDir}/testSnapshotFolder")
+    testSnapshotFolder.mkdirs()
+    setting 'path.repo', "${buildDir}/testSnapshotFolder"
 }
 
 task integTestRemote(type: RestIntegTestTask) {

--- a/src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java
+++ b/src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java
@@ -17,7 +17,6 @@ import org.apache.commons.lang.StringUtils;
 import org.apache.http.client.utils.URIBuilder;
 import org.apache.http.util.EntityUtils;
 import org.opensearch.Version;
-import org.opensearch.common.xcontent.support.XContentMapValues;
 import org.opensearch.core.common.bytes.BytesReference;
 import org.opensearch.common.xcontent.XContentHelper;
 import org.opensearch.core.xcontent.DeprecationHandler;
@@ -74,8 +73,6 @@ import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
-import static org.hamcrest.Matchers.hasSize;
-import static org.hamcrest.Matchers.notNullValue;
 import static org.opensearch.knn.common.KNNConstants.DIMENSION;
 import static org.opensearch.knn.common.KNNConstants.ENCODER_PARAMETER_PQ_CODE_SIZE;
 import static org.opensearch.knn.common.KNNConstants.ENCODER_PARAMETER_PQ_M;
@@ -1986,17 +1983,7 @@ public class KNNRestTestCase extends ODFERestTestCase {
 
     @SneakyThrows
     protected void setupSnapshotRestore(String index, String snapshot, String repository) {
-        Request clusterSettingsRequest = new Request("GET", "/_cluster/settings");
-        clusterSettingsRequest.addParameter("include_defaults", "true");
-        Response clusterSettingsResponse = client().performRequest(clusterSettingsRequest);
-        Map<String, Object> clusterSettings = entityAsMap(clusterSettingsResponse);
-
-        @SuppressWarnings("unchecked")
-        List<String> pathRepos = (List<String>) XContentMapValues.extractValue("defaults.path.repo", clusterSettings);
-        assertThat(pathRepos, notNullValue());
-        assertThat(pathRepos, hasSize(1));
-
-        final String pathRepo = pathRepos.get(0);
+        final String pathRepo = System.getProperty("tests.path.repo");
 
         // create index
         createIndex(index, getDefaultIndexSettings());

